idaes/power_generation/unit_models/heat_exchanger_3streams.py

```diff
@@ -0,0 +1,551 @@
+            self.deltaP_side_1 = Reference(self.side_1.deltaP)
+            self.deltaP_side_2 = Reference(self.side_2.deltaP)
+            self.deltaP_side_3 = Reference(self.side_3.deltaP)
+
+        # Performance parameters and variables
+        # Temperature driving force
+        self.temperature_driving_force_side_2 = Var(self.flowsheet().config.
+                                                    time,
+                                                    initialize=1.0,
+                                                    doc='Mean driving force '
+                                                    'for heat exchange')
+
+        # Temperature driving force
+        self.temperature_driving_force_side_3 = Var(self.flowsheet().
+                                                    config.time,
+                                                    initialize=1.0,
+                                                    doc='Mean driving force '
+                                                    'for heat exchange')
+
+        # Temperature difference at side 2 inlet
+        self.side_2_inlet_dT = Var(self.flowsheet().config.time,
+                                   initialize=1.0,
+                                   doc='Temperature difference '
+                                   'at side 2 inlet')
+
+        # Temperature difference at side 2 outlet
+        self.side_2_outlet_dT = Var(self.flowsheet().config.time,
+                                    initialize=1.0,
+                                    doc='Temperature difference '
+                                    'at side 2 outlet')
+
+        # Temperature difference at side 3 inlet
+        self.side_3_inlet_dT = Var(self.flowsheet().config.time,
+                                   initialize=1.0,
+                                   doc='Temperature difference'
+                                   ' at side 3 inlet')
+
+        # Temperature difference at side 3 outlet
+        self.side_3_outlet_dT = Var(self.flowsheet().config.time,
+                                    initialize=1.0,
+                                    doc='Temperature difference '
+                                    'at side 3 outlet')
+
+        # Driving force side 2
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Log mean temperature difference calculation")
+        def LMTD_side_2(b, t):
+            return b.temperature_driving_force_side_2[t] == \
+                ((b.side_2_inlet_dT[t]**0.3241
+                  + b.side_2_outlet_dT[t]**0.3241)/1.99996)**(1/0.3241)
+
+        # Driving force side 3
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Log mean temperature difference calculation")
+        def LMTD_side_3(b, t):
+            return b.temperature_driving_force_side_3[t] == \
+                ((b.side_3_inlet_dT[t]**0.3241
+                  + b.side_3_outlet_dT[t]**0.3241)/1.99996)**(1/0.3241)
```

Comment:
Why the weird coefficients here? Is this from that weird case where someone tried to "estimate" parameters for the Underwood approximation to LMTD? If so, you should use the Underwood approximation (1/3 and 2), as you cannot estimate these coefficients.